src/DistributedWriteAheadLog/Record.cpp

```diff
@@ -3,31 +3,46 @@
 #include <DataStreams/NativeBlockInputStream.h>
 #include <DataStreams/NativeBlockOutputStream.h>
 /// #include <DataStreams/materializeBlock.h>
+#include <Compression/CompressedReadBuffer.h>
+#include <Compression/CompressedWriteBuffer.h>
+#include <Compression/CompressionFactory.h>
 #include <IO/ReadBufferFromMemory.h>
 #include <IO/ReadHelpers.h>
 #include <IO/WriteBufferFromVector.h>
 #include <IO/WriteHelpers.h>
 
 namespace DWAL
 {
-ByteVector Record::write(const Record & record)
+ByteVector Record::write(const Record & record, bool compressed)
```

Comment:
Before:

```
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from CheckRecordSerializationDeserialization
[ RUN      ] CheckRecordSerializationDeserialization.Benchmark
write: 809017 read: 58187
[       OK ] CheckRecordSerializationDeserialization.Benchmark (18435 ms)
[----------] 1 test from CheckRecordSerializationDeserialization (18435 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (18436 ms total)
[  PASSED  ] 1 test.
```

Reply:
After:
root@localhost:/data/kchen/daisy/static-clang-release# src/unit_tests_dbms  --gtest_filter=CheckRecordSerializationDeserialization.WriteBenchmark
Running main() from ../contrib/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = CheckRecordSerializationDeserialization.WriteBenchmark
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from CheckRecordSerializationDeserialization
[ RUN      ] CheckRecordSerializationDeserialization.WriteBenchmark
write uncompressed: 879090/s write compressed: 183212/s
[       OK ] CheckRecordSerializationDeserialization.WriteBenchmark (6609 ms)
[----------] 1 test from CheckRecordSerializationDeserialization (6609 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (6610 ms total)
[  PASSED  ] 1 test.

Reply:
We almost have 10 % perf improvement for uncompressed write

Reply:
After (read):

```
root@localhost:/data/kchen/daisy/static-clang-release# src/unit_tests_dbms  --gtest_filter=CheckRecordSerializationDeserialization.ReadBenchmark
Running main() from ../contrib/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = CheckRecordSerializationDeserialization.ReadBenchmark
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from CheckRecordSerializationDeserialization
[ RUN      ] CheckRecordSerializationDeserialization.ReadBenchmark
read uncompressed: 58747/s read compressed: 51990/s
[       OK ] CheckRecordSerializationDeserialization.ReadBenchmark (36270 ms)
[----------] 1 test from CheckRecordSerializationDeserialization (36270 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (36271 ms total)
[  PASSED  ] 1 test.
```

Reply:
For read, we don't have perf degradation for uncompressed read